File: AIR-ARR/Programowanie Obiektowe/11/prj/src/stos.cpp
#include "stos.hh"

/*!
 *\file
 *\brief	Plik zawierający definicje metody z klasy Stos.
*/

void Stos::drukuj() {
	if (rozmiar() > 0) {
		cout << endl << "Stos:" << endl;
		Stos pom = (*this);
		while (pom.rozmiar() > 0) {
			cout << pom.szczyt() << endl;
			pom.usun();
		}
	}
	else cout << "Stos jest pusty." << endl;
}

File: INF/Algorytmy i Struktury Danych/zad3/jeden/Stos.java
package jeden;

public class Stos <R> {
	public Element<R> wierzcholek;
	
	public Stos() {
		wierzcholek = null;
	}
	
	public Element<R> getW() {
		return wierzcholek;
	}
	
	public void push(R e) {
		Element<R> pom = new Element<R>(e,wierzcholek);
		wierzcholek = pom;
	}
	
	public Element<R> pop() {
		Element<R> pom = null;
		if (wierzcholek != null) {
			pom = wierzcholek;
			wierzcholek = wierzcholek.getNast();
		}
		else System.out.println("Stos jest pusty.");
		return pom;
	}
}